MAM-3990-images-in-akkoma-posts-have-wrong-dimensions

derspyy opened the pull request:
(there is an issue with the diff where an indentation change looks horrible)

this addresses the issue where some services omit image metadata (due to not doing background processing).
affects: akkoma users & mastodon users viewing akkoma profiles or replies.

we need to see if this breaks anything, i made images without metadata fallback to 16:9 with a fit scale. this is how they look:

<img width="314" alt="image" src="https://github.com/TheBLVD/mammoth/assets/65043391/454d936d-0e67-444b-9c3d-3a581e45ab0a">

this is how they compare to akkoma's frontend:

<img width="611" alt="image" src="https://github.com/TheBLVD/mammoth/assets/65043391/a2378d05-1239-41e9-9095-fb89703bf29d">

mastodon images should be untouched.

derspyy commented on Mammoth/Views/Cells/PostCardCell/PostCardImage.swift:
```diff
@@ -285,57 +285,57 @@ final class PostCardImage: UIView {
             }
+            let ratio = self.media?.meta?.original?.aspect ?? aspect ?? 16.0 / 9.0
     
-            if let ratio = self.media?.meta?.original?.aspect {
```
just noticed that the diff here is horrible. i just changed this first line, the rest is indentation.

derspyy commented on Mammoth/Views/Cells/PostCardCell/PostCardVideo.swift:
```diff
@@ -518,58 +518,58 @@ final class PostCardVideo: UIView {
                 }
+                let ratio = self.media?.meta?.original?.aspect ?? aspect ?? (self.media?.type == .audio ? 1.0 : 16.0 / 9.0)
         
-                if let ratio = self.media?.meta?.original?.aspect {
```
same here

bnolens reviewed (changes requested):
There's an issue where the blurhash image is not the same size as the final image. If that's fixed, I think looks good. 👍 
https://github.com/TheBLVD/mammoth/assets/221925/ee33b753-957c-4b78-80fb-ffc6c8aca467

derspyy commented:
i really don't think this is related to my pr (which only concerns images with no metadata and therefore no blurhash), but i can also fix that if you want

bnolens commented:
> i really don't think this is related to my pr (which only concerns images with no metadata and therefore no blurhash), but i can also fix that if you want

What about this change you made? https://github.com/TheBLVD/mammoth/pull/485/files#diff-db60f1901e92e5866bdbe16565649da7b8f6e6b3c4888169a3ac93faad313f6cR26

derspyy commented:
> > i really don't think this is related to my pr (which only concerns images with no metadata and therefore no blurhash), but i can also fix that if you want
> 
> 
> 
> What about this change you made? https://github.com/TheBLVD/mammoth/pull/485/files#diff-db60f1901e92e5866bdbe16565649da7b8f6e6b3c4888169a3ac93faad313f6cR26

for reference: since i changed the images to FIT instead of FILL, the placeholders with wrong dimensions became more visible. fixed it though

derspyy commented:
made it so that only these images will be set to FIT scale.

bnolens reviewed (approved):
LGTM
